rhiever/tpot
# Bug fixes improvements

## What does this PR do?
Some minor bug fixes and robustness improvements.
Split from the 


## Where should the reviewer start?
Each commit is a self contained change/fix, can be reviewed independently.
The retries commit should be viewed ignoring whitespace, that will reduce the diff by ~90%, The try + except clause just indented the lines it surrounded.


## How should this PR be tested?
You might notice changes in a few scenarios:
1. supply a bad data set and see that you get a meaningful exception when running tpot (what exception a model had trying to .fit the data).
2. run tpot and see argument list printed correctly: arg_name == arg_value VS index == arg_name
3. run tpot and after optimizing a single pipeline ctrl+C 3-4 times to see tpot doesn't give up and still outputs an optimized pipeline.


## Any background context you want to provide?
These fixes are part of a bigger split pull request which will include 2-3 more pull requests for each separate feature.


## What are the relevant issues?

## Screenshots (if appropriate)



## Questions:

- Do the docs need to be updated? No
- Does this PR add new (Python) dependencies? No
